website/addons/box/utils.py

```diff
@@ -78,6 +86,40 @@ def log(self, action, extra=None, save=False):
             self.node.save()
 
 
+def refresh_oauth_key(external_account, force=False):
+    """If necessary, refreshes the oauth key associated with
+    the external account.
+    """
+    if external_account.expires_at is None and not force:
+        return
+
+    if force or (external_account.expires_at - datetime.utcnow()).total_seconds() < settings.REFRESH_TIME:
+        key = refresh_v2_token(settings.BOX_KEY, settings.BOX_SECRET, external_account.refresh_token)
+
+        external_account.oauth_key = key['access_token']
+        external_account.refresh_token = key['refresh_token']
+        external_account.expires_at = datetime.utcfromtimestamp(time.time() + key['expires_in'])
+        external_account.save()
+
+
+def get_folders(client):
+    """Gets a list of folders in a user's Box, including the root.
+    Each folder is represented as a dict with its display name and path.
+    """
+    metadata = client.metadata('/', list=True)
+    serializer = BoxSerializer
+    # List each folder, including the root
+    root = {
+        'name': '/ (Full Box)',
+        'path': '',
+    }
+    folders = [root] + [
+        serializer.serialize_folder(each)
+        for each in metadata['contents'] if each['is_dir']
+    ]
+    return folders
+
+
 def handle_box_error(error, msg):
```

Comment:
It looks like all usages of `handle_box_error` have been deleted -- can this function be removed as well?